src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Appearance/WindowBackdrop.cs

```diff
@@ -217,67 +163,20 @@ internal static bool RemoveBackground(System.Windows.Window window)
         return true;
     }
 
-    internal static bool EnableGlassFrame(System.Windows.Window window, WindowBackdropType backdropType)
-    {
-        if (window is null)
-        {
-            return false;
-        }
-
-        if (window.IsLoaded)
-        {
-            IntPtr windowHandle = new WindowInteropHelper(window).Handle;
-
-            if (windowHandle == IntPtr.Zero)
-            {
-                return false;
-            }
-            return EnableGlassFrame(windowHandle, backdropType);
-        }
-
-        window.Loaded += (sender, _) =>
-        {
-            IntPtr windowHandle =
-                new WindowInteropHelper(sender as System.Windows.Window)?.Handle ?? IntPtr.Zero;
-
-            if (windowHandle == IntPtr.Zero)
-            {
-                return;
-            }
-
-            EnableGlassFrame(windowHandle, backdropType);
-        };
-
-        return true;
-    }
-
-    private static bool EnableGlassFrame(IntPtr hWnd, WindowBackdropType backdropType)
+    private static bool UpdateGlassFrame(IntPtr hWnd, WindowBackdropType backdropType)
     {
         if (hWnd == IntPtr.Zero)
         {
             return false;
         }
 
-        if (!NativeMethods.IsWindow(hWnd))
+        MARGINS margins = new MARGINS();
+        if(backdropType != WindowBackdropType.None)
         {
-            return false;
+            margins = new MARGINS { cxLeftWidth = -1, cxRightWidth = -1, cyTopHeight = -1, cyBottomHeight = -1 };                    
```

Comment:
This enables the glass frame effect in windows. Also, now I understood your query about WBT.None 
 
Well the operation that I am doing for None is just resetting the glass frame. Now, your query was why do that operation if backdrop is None, so to answer that, this call will have no effect on existing state of window if it's glass frame was not modified earlier. 

Say, you opened a Window, and we get to the point where we call UpdateGlassFrame. Now, for the none case, the operations have net zero effect, however for other backdrop type, it will create the glass effect.

Now, suppose we modified the backdrop from mica to none, in that case we want to remove the glass frame and hence we do the reset call ( or the operations that are being done there ).